libyul/optimiser/SSATransform.cpp

```diff
@@ -165,24 +165,24 @@ class IntroduceControlFlowSSA: public ASTModifier
 	NameDispenser& m_nameDispenser;
 	std::set<YulString> const& m_variablesToReplace;
 	/// Variables (that are to be replaced) currently in scope.
-	std::set<YulString> m_variablesInScope;
-	/// Set of variables that do not have a specific value.
-	std::set<YulString> m_variablesToReassign;
+	std::vector<YulString> m_variablesInScope;
```

Comment:
Why does `m_variablesInScope` need to be a vector? I don't see it being iterated anywhere. We only use it for lookups.

Reply:
Iterating by itself is not a problem - `+=` and `-=` ops are also a part of the problem, so I changed all of them to be safe; I can revert it back to a set and see if it has any negative effect, although not sure how worth it that would be.

Reply:
How is iterating not a problem? It's the main thing that produces non-determinism here, isn't it?

`+=` and `-=` on the other hand should never be a problem on their own. They may affect order of items in the the container but that's fine if we never depend on that order. And simple lookups don't.

> although not sure how worth it that would be.

I don't really like the idea of shooting in the dark here. I'd rather only change the bits that are known to produce this non-determinism.

Reply:
Ah you're right, I thought `m_variablesInScope` was assigned to another container that was iterated, but this isn't the case, so `m_variablesInScope` should be safe to revert back to `std::set`.

Reply:
Yup, you were right; `m_variablesInScope` can be a set - I've reverted it back, pushed the changes, and re-ran the repro to makes sure it's fine.